mobile_config/src/gateway_info.rs

```diff
@@ -369,15 +369,34 @@ pub(crate) mod db {
             join key_to_assets kta on infos.asset = kta.asset
         "#;
     const BATCH_SQL_WHERE_SNIPPET: &str = " where kta.entity_key = any($1::bytea[]) ";
-    const DEVICE_TYPES_AND_SNIPPET: &str = " and device_type::text = any($2) ";
+    const DEVICE_TYPES_WHERE_SNIPPET: &str = " where device_type::text = any($1) ";
+
+    const GET_UPDATED_RADIOS: &str =
+        "SELECT entity_key FROM mobile_radio_tracker WHERE last_changed_at >= $1";
 
     lazy_static::lazy_static! {
         static ref BATCH_METADATA_SQL: String = format!("{GET_METADATA_SQL} {BATCH_SQL_WHERE_SNIPPET}");
-        static ref GET_METADATA_SQL_REFRESHED_AT: String = format!(r#"{GET_METADATA_SQL}
-            where ( infos.refreshed_at >= $1 OR (infos.refreshed_at IS NULL AND infos.created_at >= $1) ) "#);
+        static ref DEVICE_TYPES_METADATA_SQL: String = format!("{GET_METADATA_SQL} {DEVICE_TYPES_WHERE_SNIPPET}");
+    }
 
-        static ref DEVICE_TYPES_METADATA_SQL: String = format!("{} {}", *GET_METADATA_SQL_REFRESHED_AT, DEVICE_TYPES_AND_SNIPPET);
+    pub async fn get_updated_radios(
+        db: impl PgExecutor<'_>,
+        min_updated_at: DateTime<Utc>,
+    ) -> anyhow::Result<HashSet<PublicKeyBinary>> {
+        let rows: Vec<Vec<u8>> = sqlx::query_scalar(GET_UPDATED_RADIOS)
+            .bind(min_updated_at)
+            .fetch_all(db)
+            .await?;
+        let mut radios = HashSet::new();
+
+        for row in rows {
+            let entity_key_b: &[u8] = &row;
+            let entity_key = bs58::encode(entity_key_b).into_string();
+            let pk = PublicKeyBinary::from_str(&entity_key)?;
+            radios.insert(pk);
+        }
```

Comment:
```suggestion
        let radios: HashSet<PublicKeyBinary> = sqlx::query_scalar(GET_UPDATED_RADIOS)
            .bind(min_updated_at)
            .fetch(db)
            .await
            .map(|row| {
                let entity_key_b: &[u8] = &row;
                let entity_key = bs58::encode(entity_key_b).into_string();
                PublicKeyBinary::from_str(&entity_key)
            })
            .collect::<Result<HashSet<PublicKeyBinary>, helium_crypto::Error>>()?;
```
this probably necessitates importing `StreamExt` or similar but avoids the extra allocations

Reply:
helium-crypto-rs provides a sqlx mapping for `PublicKeyBinary` https://github.com/helium/helium-crypto-rs/blob/main/src/public_key_binary.rs#L115

Reply:
Hi @michaeldjeffrey thanks for information. The helium-crypto-rs provides a sqlx mapping for PublicKeyBinary which stored in database as `string`. In our case pubkey is stored as a `bytea` type. So when I try to use helium-crypto-rs 
 mapping like below
 ```rust
 Ok(sqlx::query_as::<_, (PublicKeyBinary,)>(GET_UPDATED_RADIOS)
            .bind(min_updated_at)
            .fetch(db)
            .map_ok(|(entity_key,)| entity_key)
            .try_collect::<HashSet<_>>()
            .await?)

 ```
I've got an error: ```mismatched types; Rust type `helium_crypto::public_key_binary::PublicKeyBinary` (as SQL type `text`) is not compatible with SQL type BYTEA ```

To make it work we need to extend `decode` function  in `helium-crypto-rs` https://github.com/helium/helium-crypto-rs/blob/main/src/public_key_binary.rs#L150 to make it support bytea. I think this is not in the scope of this PR and I'm not sure that it is worth since I have no information how often we store PublicKeyBinary in bytea type.

Reply:
No doubt. That's a good @madninja question. Anyways, carry on 👯